rust/abacus-base/src/settings/chains.rs

```diff
@@ -23,16 +26,33 @@ impl Default for ChainConf {
     }
 }
 
+/// Addresses for outbox chain contracts
+#[derive(Clone, Debug, Deserialize, Default)]
+pub struct OutboxAddresses {
+    /// Address of the Outbox contract
+    pub outbox: String,
+}
+
+/// Addresses for inbox chain contracts
+#[derive(Clone, Debug, Deserialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct InboxAddresses {
+    /// Address of the Inbox contract
+    pub inbox: String,
+    /// Address of the InboxValidatorManager contract
+    pub validator_manager: String,
+}
+
 /// A chain setup is a domain ID, an address on that chain (where the outbox or
 /// inbox is deployed) and details for connecting to the chain API.
 #[derive(Clone, Debug, Deserialize, Default)]
-pub struct ChainSetup {
+pub struct ChainSetup<T> {
```

Comment:
I think it's nice having the addresses alongside the ChainConf in _some way_ so that a chain's config can be all in one place without any merging of objects. Maybe I'm not understanding what you mean tho, I mean that it's kinda nice to do this:
```
{
  "inboxes": {
    "kovan": {
      "domain": "3000",
      "name": "kovan",
      "rpcStyle": "ethereum",
      "connection": {
        "type": "http",
        "url": ""
      },
      "addresses": {
        "inbox": "0x84eA74d481Ee0A5332c457a4d796187F6Ba67fEB",
        "validatorManager": "0x67d269191c92Caf3cD7723F116c85e6E9bf55933"
      }
    }
  }
}
```

Instead of:
```
{
  "inboxes": {
    "kovan: {
        "inbox": "0x84eA74d481Ee0A5332c457a4d796187F6Ba67fEB",
        "validatorManager": "0x67d269191c92Caf3cD7723F116c85e6E9bf55933"
    }
  },
  "chains": {
    "kovan" {
      "domain": "3000",
      "name": "kovan",
      "rpcStyle": "ethereum",
      "connection": {
        "type": "http",
        "url": ""
      },
    }
  }
}
```
Because you don't have to worry about having an `inboxes.kovan` section but not a `chains.kovan` section.

I think if we ever get to a point where we can't get away with the first example and reference a chain in a bunch of spots, then moving to the second example is worth it though